RecoHI/HiTracking/plugins/HIBestVertexProducer.cc

```diff
@@ -53,68 +55,93 @@ void HIBestVertexProducer::produce
   if(vertices1->size()==0)
     LogError("HeavyIonVertexing") << "adaptive vertex collection is empty!" << endl;
 
-  if(vertices1->begin()->zError()<3) { 
-  
-    reco::VertexCollection::const_iterator vertex1 = vertices1->begin();
-    newVertexCollection->push_back(*vertex1);
-
-    LogInfo("HeavyIonVertexing") << "adaptive vertex:\n vz = (" 
-				 << vertex1->x() << ", " << vertex1->y() << ", " << vertex1->z() << ")" 
-				 << "\n error = ("
-				 << vertex1->xError() << ", " << vertex1->yError() << ", " 
-				 << vertex1->zError() << ")" << endl;
-  } else {
-    
-    //** Get fast median vertex **/
-    edm::Handle<reco::VertexCollection> vc2;
-    ev.getByToken(theMedianVertexCollection, vc2);
-    const reco::VertexCollection * vertices2 = vc2.product();
-    
-    //** Get beam spot position and error **/
-    reco::BeamSpot beamSpot;
-    edm::Handle<reco::BeamSpot> beamSpotHandle;
-    ev.getByToken(theBeamSpotTag, beamSpotHandle);
-
-    if( beamSpotHandle.isValid() ) 
-      beamSpot = *beamSpotHandle;
-    else
-      LogError("HeavyIonVertexing") << "no beamspot found "  << endl;
+//** Final vertex collection if needed **//
+  bool hasFinalVertex = false;
+  if(theUseFinalAdapativeVertexCollection){
+    edm::Handle<reco::VertexCollection> vc0;
+    ev.getByToken(theFinalAdaptiveVertexCollection, vc0);
+    const reco::VertexCollection *vertices0 = vc0.product();
+    if(vertices0->size()==0)
+      LogError("HeavyIonVertexing") << "final adaptive vertex collection is empty!" << endl;
+
+  //if using final vertex and has a good vertex, use it
+    if(vertices0->begin()->zError()<3){
```

Comment:
should this magic number of 3 be made configurable?
Since now it is needed in two places in this file, please define a named constant and use it in both places.